database/mysql_testdb.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/moov-io/base/docker"
+	"github.com/moov-io/base/log"
+	"github.com/ory/dockertest"
+)
+
+var (
+	mysqlInstance *MySQLTestDB = &MySQLTestDB{}
+)
+
+func init() {
+	// after the above mysqlInstance is initialized with a zero-value
+	// write-lock our mysqlInstance so tests trying to read (RLock()
+	// are prevented until the underlying container is started.
+	mysqlInstance.mu.Lock()
+
+	go func() {
+		// ugly hack to stall .Wait() and allow tests to startup
```

Comment:
A blocking channel could work well here. Or a using another `sync.WaitGroup` to block and calling `wg.Done()` in the `once.Do()`